app/src/main/java/com/nutomic/syncthingandroid/activities/SettingsActivity.java

```diff
@@ -253,6 +276,30 @@ public void onActivityCreated(Bundle savedInstanceState) {
             } catch (PackageManager.NameNotFoundException e) {
                 Log.d(TAG, "Failed to get app version name");
             }
+
+            openSubPrefScreen(screen);
+        }
+
+        private void openSubPrefScreen(PreferenceScreen prefScreen) {
+            Bundle bundle = getArguments();
+            if (bundle == null) {
+                return;
+            }
+            String openSubPrefScreen = bundle.getString(EXTRA_OPEN_SUB_PREF_SCREEN, "");
+            // Open sub preferences screen if EXTRA_OPEN_SUB_PREF_SCREEN was passed in bundle.
+            if (openSubPrefScreen != null && !TextUtils.isEmpty(openSubPrefScreen)) {
+                Log.v(TAG, "Transitioning to pref screen " + openSubPrefScreen);
+                PreferenceScreen categoryRunConditions = (PreferenceScreen) findPreference(openSubPrefScreen);
```

Comment:
It could be null as all the other stuff having a constant referring to the xml layout. As I already looked for a solution to directly refer to R.layout resources by names I discovered android java doesn't have a function/class for doing it. If one xml part is renamed and the constant in code not, the app would in this case crash when trying to open the main pref screen.

Reply:
So answer to the question: Currently never null, if someone does not code correct, could get null immediately noticed when clicking through the UI during test.